zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/value/JobListenerEventType.java

```diff
@@ -36,15 +36,16 @@ public enum JobListenerEventType {
   END,
 
   /**
-   * Represents the `complete` event for a task listener. This event type is used to indicate that
-   * the listener should be triggered after a user task complete operation was invoked.
+   * Represents the `completing` event for a task listener. This event type is used to indicate that
+   * the listener should be triggered when a user task is completing. It allows to execute custom
+   * logic before the task is completed, to correct user task data, and to deny the completion.
    */
-  COMPLETE,
+  COMPLETING,
 
   /**
-   * Represents the `assignment` event for a task listener. This event type is used to indicate that
-   * the listener should be triggered when a user task is assigned, claimed (assigned to the current
-   * user), or unassigned.
+   * Represents the `assigning` event for a task listener. This event type is used to indicate that
+   * the listener should be triggered when a user task is assigning. It allows to execute custom
+   * logic before the task is assigned, to correct user task data, and to deny the assignment.
    */
-  ASSIGNMENT
+  ASSIGNING,
```

Comment:
❓ What do you think about also renaming the [UserTaskIntent-s](https://github.com/camunda/camunda/blob/1df25b546f55b29f3f007b82390a180103e5c8de/zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/intent/UserTaskIntent.java#L73-L87) to align with the updated task listener event types?

For example:
```
ASSIGNMENT_DENIED -> ASSIGNING_DENIED  
COMPLETION_DENIED -> COMPLETING_DENIED
```

This would make the naming consistent in `UserTaskIntent` also and improve the readability of the records flow:

```java
C ASSIGN  
E ASSIGNING  
// If the assigning listener denies the operation  
C DENY_TASK_LISTENER  
// Then we would have
E ASSIGNING_DENIED
```

Reply:
And adjust event applier names:
```scala
UserTaskAssignmentDeniedApplier -> UserTaskAssigningDeniedApplier
UserTaskCompletionDeniedApplier -> UserTaskCompletingDeniedApplier
```

Reply:
> ASSIGNING and COMPLETING can be considered as not having been denied because those events have actually happened and were emitted.

Thanks @korthout for highlighting this point — in that case, `ASSIGNMENT_DENIED` and `COMPLETION_DENIED` would suit better.

At the same time, later we’ll might have `UPDATE_DENIED` (according to [**Options 4**](https://camunda.slack.com/archives/C0622NWC9LK/p1728996247956779)), which makes it clear that finding the perfect naming convention isn’t straightforward! :see_no_evil:😸

Reply:
💭 `ASSIGNMENT_DENIED` and `COMPLETION_DENIED` would suit better here I think 👍